sktm/reporter.py

```diff
@@ -0,0 +1,322 @@
+    def __init__(self, new_name, file_path):
+        """
+        Create mail attachment.
+
+        Args:
+            new_name:  Name to use for the attachment, excluding suffix.
+            file_path: Absolute path to the file that should be attached.
+        """
+        self.filename = new_name
+        self.data = self.__mime_data(file_path, file_path.endswith('.gz'))
+
+    def __repr__(self):
+        return self.filename
+
+    def __mime_data(self, file_path, compressed):
+        """
+        Wrap the data from passed file into MIMEText or MIMEApplication
+        attachment, based on its type.
+
+        Args:
+            file_path:  Absolute path to the file containing the data.
+            compressed: True if the data was compressed, False if not.
+
+        Returns:
+            Resulting MIMEText or MIMEApplication object containing the data.
+        """
+        with open(file_path, 'r') as datafile:
+            data = datafile.read()
+
+        if compressed:
+            attachment = MIMEApplication(data)
+        else:
+            attachment = MIMEText(data, _charset='utf-8')
+
+        attachment.add_header('content-disposition',
+                              'attachment',
+                              filename=self.filename)
+        return attachment
+
+
+class MailReporter(object):
+    def __init__(self, assets_dir, email_from, email_to, report_intro,
+                 report_footer, smtp_url=None, jenkins_job_ids=None,
+                 headers=[]):
+        """
+        Initialize the mail report.
+
+        Args:
+            assets_dir:      Directory to retrieve test results from.
+            email_to:        List of emails to send the report to.
+            email_from:      Sender of the message. If None, defaults to
+                             'Kernel-CI kernel-ci@localhost'.
+            report_intro:    Path to the file containing text that the report
+                             should be prefixed with (report introduction).
+            report_footer:   Path to the file containing text that the report
+                             should ended with.
+            smtp_url:        URL to use as SMTP server, if None localhost is
+                             used.
+            jenkins_job_ids: List of Jenkins job IDs to add as a custom header.
+            headers:         List of headers to add to the report.
+        """
+        if assets_dir:
+            self.assets_dir = os.path.abspath(assets_dir)
+        else:
+            raise Exception('--assets directory not provided!')
+
+        self.report = MIMEMultipart()
+        self.report['From'] = (email_from
+                               if email_from
+                               else 'Kernel-CI kernel-ci@localhost')
+        self.report['To'] = ', '.join([recipient for recipient in email_to])
+        for header in headers:
+            key, value = header.split(':', 1)
+            self.report[key] = value
+
```

Comment:
Who sanitizes the input?  If I type --header foo --header bar:blah, I think things will go bad?  Or is the input files verified?
Not that many non-scripts will touch those cli options, but in case things get fat fingered..

Reply:
This logic is taken from skt (but can be changed if needed). Missing space as in your second example is not a problem, only missing `:` is (as per your first example). The format is described in the help string if someone wants to use the command manually. IMHO, things should "go bad" if you pass invalid option. The error thrown should show the line which tries to split the header so it should be obvious what is the problem but if you would feel more comfortable with it, I can add a custom error message saying that the header passed doesn't conform to the required format.

Reply:
Ok.  If this is more-or-less a copy-n-paste of skt, then it is probably fine.  Eventually, it would be nice to have some checking going on to avoid silly bugs.  I wonder if a simple 'if not value: raise Exception()' works here.

Reply:
Yeah.  thanks.